autoload/ale/completion.vim

```diff
@@ -999,6 +1008,10 @@ function! ale#completion#Queue() abort
 
     call ale#completion#StopTimer()
 
+    " Close the previous completion menu (if any), so that the newer autocompletion
+    " candidates will show up
+    call s:closePopupMenu()
```

Comment:
I did some tests and the patch works fine (at least does not breaks anything) but I think this can be simplified.

You can call  `s:closePreviousCompletionMenu()` directly here instead of `s:closePopupMenu()` and get the same result. I tested this and works fine.  You can move the comment to the s:closePreviousCompletionMenu().

With this change you can completely remove `s:closePopupMenu()` and `s:SetupClosePopupMenu()` functions simplifying the code.

Reply:
> You can call `s:closePreviousCompletionMenu()` directly here instead of `s:closePopupMenu()` and get the same result. I tested this and works fine. 

Unfortunately, this is not the case. I just tried this on vim8.0 and, despite the abort present in the definition of ```s:closePreviousCompletionMenu()```, vim still barfs 4 errors complaining that ```complete_info``` does not exist etc. This is the same problem that the first version of the patch had.

> I did some tests and the patch works fine (at least does not breaks anything) but I think this can be simplified.
>
>  <snip...>
> You can move the comment to the s:closePreviousCompletionMenu(). 
>
>  <snip...>
> With this change you can completely remove `s:closePopupMenu()` and `s:SetupClosePopupMenu()` functions simplifying the code.

In order to remove `s:SetupClosePopupMenu()` and simplify things, we'll have to pull the `exists('*complete_info')` check inside `s:closePreviousCompletionMenu()`, meaning it'll be performed every time autocompletion takes place instead of only once during setup. I won't argue about performance overheads (esp. without hard numbers in hand) as I can see the value of simpler code.

I'll push an additional commit with the simplified code in this branch for review purposes, with the intention to squash it just before merge.

Reply:
Added commit, rebased on latest master.